### PR TITLE
Fix random number generator

### PR DESCRIPTION
**Pull request by evanvosberg**

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -10,23 +10,21 @@ var CryptoJS = CryptoJS || (function (Math, undefined) {
      *
      * As Math.random() is cryptographically not safe to use
      */
-    var secureRandom = function () {
+    var cryptoSecureRandomInt = function () {
         // Native crypto module on NodeJS environment
         try {
-            // Crypto from global object
-            var crypto = global.crypto;
+            // Native rypto from global object or import via require
```
Comment (wartab) [inline src/core.js]: typo in crypto

File: src/core.js [hunk 2]
```diff
@@ -10,23 +10,21 @@ var CryptoJS = CryptoJS || (function (Math, undefined) {
...
         } catch (err) {}
 
         // Native crypto module in Browser environment
         try {
             // Support experimental crypto module in IE 11
             var crypto = window.crypto || window.msCrypto;
 
-            // Create a random float number between 0 and 1
-            return Number('0.' + window.crypto.getRandomValues(new Uint32Array(1))[0]);
+            return (crypto.getRandomValues(new Uint32Array(1))[0]) | 1;
```
Comment (wartab) [inline src/core.js]: Why `| 1`? That will make it so the returned number will always be odd, therefore making this not a 32 bit random number but only 31.

File: src/core.js [hunk 3]
```diff
@@ -321,7 +319,7 @@ var CryptoJS = CryptoJS || (function (Math, undefined) {
             var words = [];
 
             for (var i = 0; i < nBytes; i += 4) {
-                words.push((secureRandom() * 0x100000000) | 0);
+                words.push((cryptoSecureRandomInt());
```
Comment (wartab) [inline src/core.js]: That's a synax error, you forgot to remove a `(`.

Comment (evanvosberg) [reply src/core.js]: Right, I did not run the code yet ...

File: src/core.js [hunk 4]
```diff
@@ -10,23 +10,21 @@ var CryptoJS = CryptoJS || (function (Math, undefined) {
...
         try {
-            // Crypto from global object
-            var crypto = global.crypto;
+            // Native rypto from global object or import via require
+            var crypto = global.crypto || require('crypto');
 
-            // Create a random float number between 0 and 1
-            return Number('0.' + crypto.randomBytes(3).readUIntBE(0, 3));
+            return crypto.randomBytes(4).readInt32LE();
         } catch (err) {}
```
Comment (wartab) [inline src/core.js]: Please check gracefully for the environment. 
Besides that I'm unaware of why crypto should be available in global in NodeJs. Only use require if require exists.
Exception management is expensive.

Comment (evanvosberg) [reply src/core.js]: crypto is available on global in NodeJS. I don't know why, but it just is.

But I agree to add checks.

**Comment by evanvosberg** [comment]
@wartab can you do a second round?

**Pull request closed without merging**